theseus/optimizer/linear/tests/test_dense_solver.py

```diff
@@ -47,9 +52,10 @@ def test_dense_solvers():
         # Add damping (using loop to make it more visually explicit)
         for i in range(batch_size):
             for j in range(n):
-                AtA_ellipsoidal_damp[i, j, j] *= 1 + damping
+                damp = damping if isinstance(damping, float) else damping[i]
```

Comment:
In the case when damping is scalar/float would it make sense to do a `as_tensor` on it and then allow it to be broadcasted? Would that make it more complicated than simplifying things?

Reply:
Do you mean for the unit test?

Reply:
I meant in the main implementation (I just realized I flagged this from a test file).

Reply:
Changed it in the main implementation. Prefer to leave as is in the test, to make sure passing a float also works.